presto-hive/src/main/java/io/prestosql/plugin/hive/PartitionAndStatementId.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.hive;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.airlift.json.JsonCodec;
+
+import java.util.Objects;
+
+import static java.util.Objects.requireNonNull;
+
+public class PartitionAndStatementId
+{
+    private final String partitionName;
+    private final int statementId;
+    private final long rowCount;
+    private final String deleteDeltaDirectory;
+
+    @JsonCreator
+    public PartitionAndStatementId(@JsonProperty("partitionName") String partitionName, @JsonProperty("statementId") int statementId, @JsonProperty("rowCount") long rowCount, @JsonProperty("deleteDeltaDirectory") String deleteDeltaDirectory)
+    {
+        this.partitionName = requireNonNull(partitionName, "partitionName is null");
+        this.statementId = statementId;
+        this.rowCount = rowCount;
+        this.deleteDeltaDirectory = deleteDeltaDirectory;
+    }
+
+    @JsonProperty("partitionName")
```

Comment:
We don't normally provide the name if it's the same as the default (all of these are the default names)

Reply:
Removed.